nativelib/src/main/scala/scala/scalanative/runtime/NativeThread.scala

```diff
@@ -42,16 +44,22 @@ trait NativeThread {
 
   @alwaysinline
   final def park(): Unit =
-    park(0, isAbsolute = false)
+    if (isMultithreadingEnabled) park(0, isAbsolute = false)
+    else NativeExecutionContext.queueInternal.helpComplete()
 
   @alwaysinline
   final def parkNanos(nanos: Long): Unit = if (nanos > 0) {
-    park(nanos, isAbsolute = false)
+    if (isMultithreadingEnabled) park(nanos, isAbsolute = false)
+    else NativeExecutionContext.queueInternal.stealWork(nanos.nanos)
```

Comment:
I'm not sure if this works as expected, as `stealWork(FiniteDuration)` rounds down to the millisecond, so in a lot of situations this won't yield.

I guess this matches what's said on the javadoc, but I also see that the `WindowsThread` uses `millis.max(1).toUInt`... I don't know, maybe this is OK. 🤷

Reply:
If we use JDK as reference implementation I’d like to point that small delay is a kind of edge case.

see: https://kirill.korins.ky/articles/sleeping-at-jvm-thread-for-a-very-short-period/ where I kept notes on that investigation